RecoTracker/MkFitCMS/src/MkStdSeqs.cc

```diff
@@ -151,120 +157,149 @@ namespace mkfit {
         y[ts] = tk.y();
         z[ts] = tk.z();
         d0[ts] = tk.d0BeamSpot(bspot.x, bspot.y);
+
+        b.register_entry_safe(
+            oldPhi[ts],
+            eta[ts]);  // If one is sure values are *within* axis ranges: b.register_entry(oldPhi[ts], eta[ts]);
       }
 
-      for (int ts = 0; ts < ns; ts++) {
+      b.finalize_registration();
+
+      for (int sorted_ts = 0; sorted_ts < ns; sorted_ts++) {
+        int ts = b.m_ranks[sorted_ts];
+
         if (not writetrack[ts])
-          continue;  // Note: this speed up prevents transitive masking (possibly marginal gain).
+          continue;  //FIXME: this speed up prevents transitive masking; check build cost!
 
         const float oldPhi1 = oldPhi[ts];
         const float pos2_first = pos2[ts];
-        const float eta1 = eta[ts];
-        const float pt1 = pt[ts];
+        const float Eta1 = eta[ts];
+        const float Pt1 = pt[ts];
         const float invptq_first = invptq[ts];
 
         // To study some more details -- need EventOfHits for this
         int n_ovlp_hits_added = 0;
+        // int  n_ovlp_hits_same_module = 0;
+        // int  n_ovlp_hits_shared = 0;
+        // int  n_ovlp_tracks = 0;
 
-        for (int tss = ts + 1; tss < ns; tss++) {
-          const float pt2 = pt[tss];
+        //these two loops may be put inside the class
+        auto phi_rng = ax_phi.Rrdr_to_N_bins(oldPhi[ts], 0.08);
+        auto eta_rng = ax_eta.Rrdr_to_N_bins(eta[ts], .1);
 
-          ////// Always require charge consistency. If different charge is assigned, do not remove seed-track
-          if (charge[tss] != charge[ts])
-            continue;
+        for (auto i_phi = phi_rng.begin; i_phi != phi_rng.end; i_phi = ax_phi.next_N_bin(i_phi)) {
+          for (auto i_eta = eta_rng.begin; i_eta != eta_rng.end; i_eta = ax_eta.next_N_bin(i_eta)) {
+            const auto cbin = b.get_content(i_phi, i_eta);
+            for (auto i = cbin.first; i < cbin.end(); ++i) {
+              int tss = b.m_ranks[i];
 
-          const float thisDPt = std::abs(pt2 - pt1);
-          ////// Require pT consistency between seeds. If dpT is large, do not remove seed-track.
-          if (thisDPt > dpt_common * (pt1))
-            continue;
+              if (not writetrack[ts])
+                continue;
+              if (not writetrack[tss])
+                continue;
+              if (tss == ts)
+                continue;
 
-          const float eta2 = eta[tss];
-          const float deta2 = std::pow(eta1 - eta2, 2);
+              const float Pt2 = pt[tss];
 
-          const float oldPhi2 = oldPhi[tss];
+              ////// Always require charge consistency. If different charge is assigned, do not remove seed-track
```

Comment:
what's the reason for so many slashes in front of the comment?
```suggestion
              // Always require charge consistency. If different charge is assigned, do not remove seed-track
```